.eslintrc

```diff
@@ -7,6 +7,7 @@
     "mocha": true,
     "es6": true
   },
+  "parser": "babel-eslint",
```

Comment:
Are you sure this is necessary? I didn't see any linter errors in your last push; only a failing test. So I am bit confused here.

Reply:
As this [stackoverflow](https://stackoverflow.com/questions/51936079/eslint-how-to-ignore-parsing-error-invalid-regular-expression-flag) said, regex symbol `s` is a ecmascript 2018 feature, eslint's default parser can not parse `/***/s` to a valid AST, and can not lint it.

There's 2 ways to fix this:

1. Add `"ecmaVersion": 6` to .eslintrc, but this will cause other eslint problems related to ecam version.
2. add babel-eslint parser to parse `/***/s` .

Reply:
https://github.com/ramda/ramda/pull/3158

Reply:
Sorry I must be missing something here.

I know that #3158 introduced the `s` flag but last time I checked your pull request failed for exactly the same reason master is failing right now:

This is the only test that is failing on master:

![Screenshot 2022-01-23 at 09 01 02](https://user-images.githubusercontent.com/681975/150671602-d59f2a69-9a10-47fe-bd04-75c9f29a7cea.png)

I'm failing to see how this relates to the regex flag.

That same test used to fail on your pull request however you have now chosen to disable that test.

Reply:
What I am trying to say is that your pr didn't fail because of that regex flag you mention but because of that unit test failure.

Reply:
Ok ignore this last comment. I just figured out that adding that `babel-eslint` was intended to fix a lint issue. But it does seem that we can fix that lint error without depending on another third-party plugin, we can just set the parser options in the eslint config file. See my other comment.

Reply:
So I spent some time trying to figure out the issue with that test and I _think_ that the test simply can't work at all considering the implementation of `swap`. I opened a PR to fix the linter and tests failures so that this PR can focus on just the documentation. See #3224